src/Lorentz/Contracts/BaseDAO/Proposal.hs

```diff
@@ -4,10 +4,50 @@
 -- | Proposal related entrypoints
 module Lorentz.Contracts.BaseDAO.Proposal
   ( propose
+  , proposalMetadata
+  , vote
+  , setVotingPeriod
+  , setQuorumThreshold
+  , flush
   ) where
 
+import Universum ((*))
 import Lorentz
+import Lorentz.Contracts.BaseDAO.Management (authorizeAdmin)
 import Lorentz.Contracts.BaseDAO.Types
+import Lorentz.Contracts.BaseDAO.FA2 (doNothing)
+
+------------------------------------------------------------------------
+-- Helper
+------------------------------------------------------------------------
+
+checkIfProposalExist
+  :: forall store pm s. (NiceParameter pm, StorageC store pm)
+  => (ProposalId & store & s) :-> (Proposal pm & s)
+checkIfProposalExist = do
+  stGet #sProposals
+  ifSome nop (failCustom_ #pROPOSAL_NOT_EXIST)
+
+ensureVotingPeriodIsNotOver
+  :: forall store pm s. (NiceParameter pm, StorageC store pm)
+  => (Proposal pm & store & s) :-> s
+ensureVotingPeriodIsNotOver = do
+  toField #pStartDate
+  swap
+  stToField #sVotingPeriod; int;
```

Comment:
There are some redundant `;`s in many places. Trailing `;` is redundant.

Reply:
I have removed the unnecessary `;`.